crates/bitwarden-encoding/src/b64.rs

```diff
@@ -0,0 +1,215 @@
+use data_encoding::{Specification, BASE64};
+use serde::{Deserialize, Serialize};
+
+/// Base64 encoded data
+///
+/// Is indifferent about padding when decoding, but always produces padding when encoding.
+#[derive(Debug, Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
+#[serde(try_from = "&str", into = "String")]
+pub struct B64(Vec<u8>);
+
+impl From<Vec<u8>> for B64 {
+    fn from(src: Vec<u8>) -> Self {
+        Self(src)
+    }
+}
+impl From<&[u8]> for B64 {
+    fn from(src: &[u8]) -> Self {
+        Self(src.to_vec())
+    }
+}
+
+impl From<B64> for Vec<u8> {
+    fn from(src: B64) -> Self {
+        src.0
+    }
+}
+
+impl AsRef<[u8]> for B64 {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl From<B64> for String {
```

Comment:
Do we need this impl? I feel like the compiler will autoref B64 as needed and use the impl for `From<&B64>`